File: oop/heritage/Java/exercice2/src/Person.java
public class Person {
    private static int NextId = 1;
    private final int Id;
    String lastname;
    String firstname;

    public Person(String lastname, String firstname) {
        this.Id = NextId++;
        this.lastname = lastname;
        this.firstname = firstname;
    }

    public int getId() {
        return Id;
    }

    public String getLastname() {
        return lastname;
    }

    public void setLastname(String lastname) {
        this.lastname = lastname;
    }

    public String getFirstname() {
        return firstname;
    }

    public void setFirstname(String firstname) {
        this.firstname = firstname;
    }

    @Override
    public String toString() {
        return "Person [Id=" + Id + ", lastname=" + lastname + ", firstname=" + firstname + "]";
    }
    
}

File: oop/heritage/CPP/exercice2/src/Person.cpp
#include "Person.hpp"

int Person::nextId = 1;

Person::Person(const std::string& lastname, const std::string& firstname)
    : id(nextId++), lastname(lastname), firstname(firstname) {}

int Person::getId() const {
    return id;
}

const std::string& Person::getLastname() const {
    return lastname;
}

void Person::setLastname(const std::string& newLastname) {
    lastname = newLastname;
}

const std::string& Person::getFirstname() const {
    return firstname;
}

void Person::setFirstname(const std::string& newFirstname) {
    firstname = newFirstname;
}

std::string Person::toString() const {
    return "Person [Id=" + std::to_string(id) + ", lastname=" + lastname + ", firstname=" + firstname + "]";
}
